snoop/data/analyzers/pdf_preview.py

```diff
@@ -0,0 +1,96 @@
+"""Task to call a service that creates pdf previews for various types of documents.
+
+The service used can be found here: [[https://github.com/thecodingmachine/gotenberg]]
+"""
+
+from .. import models
+from django.conf import settings
+import requests
+from ..tasks import snoop_task, SnoopTaskBroken, returns_json_blob
+import os
+import mimetypes
+
+
+PDF_PREVIEW_MIME_TYPES = {
+    'text/plain',
+    'text/rtf',
+    'application/msword',
+    'vnd.openxmlformats-officedocument.wordprocessingml.document',
+    'application/msexcel',
+    'vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/mspowerpoint',
+    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+    'application/vnd.oasis.opendocument.presentation',
+    'application/vnd.oasis.opendocument.spreadsheet',
+    'application/vnd.oasis.opendocument.text',
+}
+"""List of mime types that the pdf generator supports.
+Based on [[https://thecodingmachine.github.io/gotenberg/#office.basic]].
+"""
+
+PDF_PREVIEW_EXTENSIONS = {
+    '.txt',
+    '.rtf',
+    '.fodt',
+    '.doc',
+    '.docx',
+    '.odt',
+    '.xls',
+    '.xlsx',
+    '.ods',
+    '.ppt',
+    '.pptx',
+    '.odp',
+}
+"""List of file extensions that the pdf generator supports.
+Based on [[https://thecodingmachine.github.io/gotenberg/#office.basic]].
+"""
+
+
+def can_create(blob):
+    """Checks if the pdf generator can process this file."""
+    if blob.mime_type in PDF_PREVIEW_MIME_TYPES:
+        return True
+
+
+def call_pdf_generator(data, filename):
```

Comment:
Do we really need filename/extension for the service to work properly? Won't it work if we send it a filename of `a` and the correct mime type?

I don't want the complication of working with filenames here, because:
- documents can appear multiple times with different extensions, what is the correct one?
- the filenames themselves can be broken bytes that come from some outdated filesystem, so they're problematic to send around -- if we really need the extension, then we need to send the filename as `'a.' + extension` instead of the original filename.